core-plugins/src/main/java/io/cdap/plugin/batch/connector/FileConnector.java

```diff
@@ -0,0 +1,144 @@
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.connector.BrowseDetail;
+import io.cdap.cdap.etl.api.connector.BrowseEntity;
+import io.cdap.cdap.etl.api.connector.BrowseEntityProperty;
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.plugin.batch.source.FileBatchSource;
+import io.cdap.plugin.format.connector.AbstractFileConnector;
+import io.cdap.plugin.format.connector.FileTypeDetector;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.PosixFileAttributes;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * File connector to browse flat file on local system
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(FileConnector.NAME)
+@Description("Flat file connector for local file system")
+@Category("Flat File")
+public class FileConnector extends AbstractFileConnector<FileConnector.FileConnectorConfig> {
+  public static final String NAME = "File";
+  static final String FILE_TYPE_KEY = "File Type";
+  static final String SIZE_KEY = "Size";
+  static final String LAST_MODIFIED_KEY = "Last Modified";
+  static final String OWNER_KEY = "Owner";
+  static final String GROUP_KEY = "Group";
+  static final String PERMISSION_KEY = "Permission";
+  private static final String DIRECTORY = "directory";
+  private static final String FILE = "file";
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest request) throws IOException {
+    String path = request.getPath();
+    File file = new File(request.getPath());
+    if (!file.exists()) {
+      throw new IllegalArgumentException(String.format("The given path %s does not exist", path));
+    }
+
+    // if it is not a directory, then it is not browsable, return the path itself
+    if (!file.isDirectory()) {
+      return BrowseDetail.builder().setTotalCount(1).addEntity(generateBrowseEntity(file)).build();
+    }
+
+    // list the files and classify them with file and directory
+    File[] files = file.listFiles();
+    if (files == null) {
+      throw new IOException(String.format("Unable to browse the path %s", path));
+    }
+
+    // sort the files by names
+    Arrays.sort(files);
+    int limit = request.getLimit() == null ? files.length : Math.min(request.getLimit(), files.length);
+    BrowseDetail.Builder builder = BrowseDetail.builder().setTotalCount(files.length);
+    for (int i = 0; i < limit; i++) {
+      // do not browse hidden files
+      if (files[i].isHidden()) {
```

Comment:
so the hidden file is not returned but counted for limit ?

Reply:
looks like hidden files do not count against the limit (which seems like the correct behavior)

Reply:
Yea, I updated yesterday to not count that towards count